### ISSUE TITLE
Does not work on 6.5.6.0

### ISSUE THREAD

**Issue by krystian-panek-wttech** [valtech/aem-easy-content-upgrade#219]
Hey! ;)

Tried to install recent AECU complete 6.4.0 on AEM 6.5.6.0 but having the following issues:

<img width="1330" alt="image" src="https://github.com/valtech/aem-easy-content-upgrade/assets/81212505/0578d748-42be-4aa9-98e0-54c083590669">

Basing on the table available here:

<img width="713" alt="image" src="https://github.com/valtech/aem-easy-content-upgrade/assets/81212505/ee50fde9-31d9-442b-a2df-0d05c257d04f">

I suppose it should work but it does not :/ 
Any help appreciated ;) 

Greetings, Krystian

**Comment by nhirrle** [comment]
Hi @krystian-panek-wttech 

Thanks for the bug report. Indeed, 6.5.3 is not supported anymore by AECU, I will go ahead and update the Readme.md.

AEM 6.5.13 is required as a a minimum https://github.com/valtech/aem-easy-content-upgrade/blob/develop/pom.xml#L493-L497 or AEM Cloud Servcie.

If you need older support you could use an old version https://github.com/valtech/aem-easy-content-upgrade/releases/tag/3.1.1 but we won't provide further updates on this one.

**Comment by krystian-panek-wttech** [comment]
Thanks for quick clarification.

I suppose because AECU depends on uber jar pointing to some specific version? Or maybe it is using some meaningful Sling API avaliable only in more recent AEM? I am asking to identify potential challenges when making it compatible with older versions of AEM... 

So for now I rather need to fork AECU and build it on my own.. :/

**Comment by nhirrle** [comment]
I'll need to cross check again but it's likely to reuse the same codebase for AEM on prem and cloud and not dealing with seperate branches. I cannot make promises right now.
SP6 is quite old, actually not even supported anymore by Adobe - 17 is the latest one. Any chance to upgrade ?

Or if not, use an old aecu version instead until AEM got upgraded?

**Comment by krystian-panek-wttech** [comment]
ok fine enough, closing

**Comment by krystian-panek-wttech** [comment]
hmm interesting @nhirrle you suggested 3.1.1 ... but I spotted that uber-jar dependency upgrade was done after 6.0.1 ... so I tried with 6.0.1 and it seems to work on 6.5.6 :)

**Comment by nhirrle** [comment]
hi @krystian-panek-wttech 
Oh indeed, I made a mistake here (checked the commit date instead of comparing tags). Good that you spotted.

**Comment by krystian-panek-wttech** [comment]
almost work... I have a problem with hook:

```
AECU InstallHook: Executed script /conf/groovyconsole/scripts/aecu/puratos/hello-world.groovy, output:

 java.lang.NoClassDefFoundError: jdk/internal/reflect/ConstructorAccessorImpl
 at java.base/jdk.internal.misc.Unsafe.defineClass0(Native Method)
 at java.base/jdk.internal.misc.Unsafe.defineClass(Unsafe.java:1192)
 at java.base/jdk.internal.reflect.ClassDefiner.defineClass(ClassDefiner.java:63)
  at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:400)
   at java.base/jdk.internal.reflect.MethodAccessorGenerator$1.run(MethodAccessorGenerator.java:394)
   at java.base/java.security.AccessController.doPrivileged(Native Method)
```

any clue?

<img width="798" alt="image" src="https://github.com/valtech/aem-easy-content-upgrade/assets/81212505/c5f1f605-ffb7-402f-a903-46ccb5f6b00e">

<img width="334" alt="image" src="https://github.com/valtech/aem-easy-content-upgrade/assets/81212505/3235d399-34ed-4a9e-b884-007efbb7debf">

**Comment by subiron** [comment]
@krystian-panek-wttech 

Hi again ;)
see: https://issues.apache.org/jira/browse/FELIX-6184
tl;dr check if 
`org.osgi.framework.bootdelegation 
`exposes:
` jdk.internal.reflect
` in your AEM instance sling.properties
BR

**Comment by krystian-panek-wttech** [comment]
no no, groovy console never needed this. it is more like classloader issue; I just restarted instance totally and it started working without doing this ;) so I can confirm it that AECU 6.0.1 works on AEM 6.5.6

**Comment by krystian-panek-wttech** [comment]
yes, you are right  @subiron ;) I missed your comment but finally found https://issues.apache.org/jira/browse/FELIX-6184
I see that @nhirrle was also struggling with that issue ;) 

pretty nasty one; thanks, guys! :) 

PS. In return for your help... and if you still need to manage on-prem AEMs you could consider using the tool that I recently created - https://github.com/wttech/aemc

**Comment by krystian-panek-wttech** [comment]
how about documenting the need for a workaround mentioned in FELIX-6184 being a prerequisite for using AEM Groovy Console and AECU? :) I lost some power here because of that thing. It would be nice to prevent wasting it by anyone else no more. I could update the readme if you wish.

**Comment by nhirrle** [comment]
Hi @krystian-panek-wttech 

Thanks for the link to the aemc tool, I will check it out further, didn't knew it yet :)

The issue in FELIX-6184 is in general with AEM 6.5 and Java 11, not related to AECU or Groovy. If you work with java 8 you don't have the issue.
Did a quick Google search and Adobe Documentation is somehow pretty hidden https://experienceleague.adobe.com/docs/experience-manager-65/deploying/deploying/troubleshooting.html?lang=en
But looks like it's mentioned as well within your company blog.
https://wttech.blog/blog/2020/aem-on-java-11-with-gradle/#3-bypass-known-felix-6184-issue

Anyhow, will update docs with the Version Infos and a hint for the sling property adoption.

**Comment by krystian-panek-wttech** [comment]
Thanks! Greetings
